Cleaned up wizard stepper

agliga opened the pull request:
## Description
~~First pass at wizard stepper cleanup. Please take a look and try it out locally to see if we want to proceed with this approach~~
Changed wizard stepper to keep `:before` and `:after` but simplified the selectors and used mixins to achieve the correct coloring

~~_**NOTE**_ This is a work in progress, once it is approved this is the path forward, will complete the rest. 
The work that needs to be done mainly is some misc cleanup in styles and docs cleanup. Also need to fix some minor spacing and some styling issues with badge version.~~
_**Should be good to go**_

## Context
Added the before/after again. Added a mixin to basically do the highlighting of the before and after of the current item. Took out the HR coloring. Only left in the three transition colors for error/info/default. The rest should be handled by normal markup.
Fixed up docs as well
~~Changed text to be absolute positioning. This does a number of things. No more `:before` or `:after` tags are needed. This causes some caveats:~~ 
 - ~~The text no longer controls the length of the container~~
 - ~~The text no longer controls the height of the container~~
~~This means that for more than 2 lines of text, it will be cut off. We can add a line clamp to show that.~~ 

~~The text max with is also fixed and can bleed into other lines. However this should be taken care of since the `min-width` of each `hr` is about `76px` and the text container is about `80px` width.~~
~~This also adds several advantages:~~
- ~~No more having to add `transition` classes. Only the `hr` will control the color.~~
- ~~No having to override different colors based on the next or transition and having to maintain spaghetti CSS (and using `!important`)~~
- ~~Is basically a full flex solution~~
- ~~No extra markup with before/after.~~

----------------
~~In case this is not the way we want to proceed and keep the old `:before/:after` we will need to add a lot of different overrides in the colors to make sure we keep the transition colors.~~

## Screenshot
<img width="1154" alt="Screen Shot 2020-04-10 at 10 24 59 AM" src="https://user-images.githubusercontent.com/1755269/79009923-91de0e00-7b15-11ea-927f-43cf4d9745ba.png">

ianmcburnie commented:
Quick comment, after a quick peek: the last letter of "Delivered" is missing. Also the text is a little too close to the icon.

<img width="345" alt="Screen Shot 2020-04-13 at 6 09 53 PM" src="https://user-images.githubusercontent.com/38065/79175262-1c429e00-7db2-11ea-828e-ca3445623ba1.png">

Oh, and I like the use of adjacent sibling selector. Very nice.

agliga commented:
> Quick comment, after a quick peek: the last letter of "Delivered" is missing. Also the text is a little too close to the icon.
> 
> <img alt="Screen Shot 2020-04-13 at 6 09 53 PM" width="345" src="https://user-images.githubusercontent.com/38065/79175262-1c429e00-7db2-11ea-828e-ca3445623ba1.png">
> 
> Oh, and I like the use of adjacent sibling selector. Very nice.

Fixed

For the text too close, I had removed the `margin-top` accidentally. Added it back
For the text cut off. So I had some `min-width`. Basically the text `min-width` has been removed. However in order to not have the text too close to one another, I added a `min-width: 10px;` to the `hr` so it preserves spacing between items.

agliga commented:
Also incorporated https://github.com/eBay/skin/pull/1078 in this PR, closing other PR

ianmcburnie commented on src/less/wizard-stepper/base/wizard-stepper.less:
```diff
@@ -18,247 +34,184 @@
-            background-color: @wizard-step-upcoming-color;
-        }
-    }
+.wizard-stepper.wizard-stepper--vertical {
```
@aglig Nitpick, but is the chained class still needed?

agliga replied:
Will remove, your right not needed.
Also will remove the nested styles
